modules/javafx.controls/src/test/java/test/javafx/scene/control/ListCellTest.java

```diff
@@ -833,5 +836,79 @@ public void testListCellHeights() {
                 cell.maxHeight(-1), 1);
     }
 
+    @Test
+    public void testChangeIndexWhileEditing_jdk_8264127() {
+        List<EditEvent> events = new ArrayList<EditEvent>();
+        list.setOnEditCancel(e -> {
+            events.add(e);
+        });
+        list.setEditable(true);
+        cell.updateListView(list);
+        cell.updateIndex(1);
+        list.edit(1);
+        cell.updateIndex(0);
+        assertTrue(!cell.isEditing());
+        assertEquals("Should still be editing 1", list.getEditingIndex(), 1);
+        assertTrue("cell re-use must trigger cancel events", events.size() == 1);
+    }
+
+    @Test
+    public void testChangeIndexToEditing1_jdk_8264127() {
+        assertChangeIndexToEditing(0, 1);
+    }
+
+    @Test
+    public void testChangeIndexToEditing2_jdk_8264127() {
+        assertChangeIndexToEditing(1, 0);
+    }
+
+    @Test
+    public void testChangeIndexToEditing3_jdk_8264127() {
+        assertChangeIndexToEditing(1, -1);
+    }
+
+    @Test
+    public void testChangeIndexToEditing4_jdk_8264127() {
+        assertChangeIndexToEditing(-1, 0);
+    }
+
+    private void assertChangeIndexToEditing(int initialCellIndex, int listEditingIndex) {
+        list.getFocusModel().focus(-1);
+        List<EditEvent> events = new ArrayList<EditEvent>();
+        list.setOnEditStart(e -> {
+            events.add(e);
+        });
+        list.setEditable(true);
+        cell.updateListView(list);
+        cell.updateIndex(initialCellIndex);
+        list.edit(listEditingIndex);
+        assertEquals("sanity: list editingIndex ", listEditingIndex, list.getEditingIndex());
+        assertFalse("sanity: cell must not be editing", cell.isEditing());
+        cell.updateIndex(listEditingIndex);
+        assertEquals("sanity: index updated ", listEditingIndex, cell.getIndex());
+        assertEquals("list editingIndex unchanged by cell", listEditingIndex, list.getEditingIndex());
+        if(listEditingIndex != -1) {
+            assertTrue(cell.isEditing());
+            assertTrue("cell re-use must trigger edit events", events.size() == 1);
+        } else {
+            // -1 represents "not editing" for the listview and "no index" for the list cell.
+            assertTrue(!cell.isEditing());
+            assertTrue("cell re-use must trigger edit events", events.size() == 0);
+        }
+    }
+
+    @Test
+    public void testUpdateCellIndexOffEditing() {
+        list.getFocusModel().focus(-1);
+        list.setEditable(true);
+        cell.updateListView(list);
+        int editingIndex = 1; // list editing
+        cell.updateIndex(editingIndex);
+        list.edit(editingIndex);
+        // here we are certain that the cell is in editing state
+        assertTrue("sanity: cell must be editing", cell.isEditing());
+        cell.updateIndex(-1); // change cell index to negative
+        assertFalse("cell must not be editing if cell index is " + cell.getIndex(), cell.isEditing());
```

Comment:
missing: assert that a cancelEvent is fired and the list editing state is unchanged (they fail until updateEditing is fixed)